openeo/extra/job_management/__init__.py

```diff
@@ -535,13 +556,23 @@ def _job_update_loop(
                     to_add = self.backends[backend_name].parallel_jobs - backend_load
                     for i in not_started.index[total_added : total_added + to_add]:
                         self._launch_job(start_job, df=not_started, i=i, backend_name=backend_name, stats=stats)
-                        stats["job launch"] += 1
-
                         job_db.persist(not_started.loc[i : i + 1])
+
+                        stats["job launch"] += 1
                         stats["job_db persist"] += 1
                         total_added += 1
 
         # Act on jobs
+        # TODO: move this back closer to the `_track_statuses` call above, once job done/error handling is also handled in threads?
+
+        while not self._result_queue.empty():
+            full_frame = job_db.get_by_status(statuses=["not_started", "created", "queued"]).copy()
+            self._process_result_queue(job_db, full_frame, stats)
+
+        with ignore_connection_errors(context="get statuses"):
+            jobs_done, jobs_error, jobs_cancel = self._track_statuses(job_db, stats=stats)
```

Comment:
I don't think we can move _track_statuses to the bottom here. Doesn't most of the stuff above depend on _track_statuses?

Reply:
then the tests have to be updated I think. 
Now it looks strange to do a bunch of  things with jobs _before_ checking their current status